OsmAnd/src/net/osmand/plus/track/helpers/GpxReaderTask.java

```diff
@@ -87,8 +87,8 @@ protected Void doInBackground(Void... voids) {
 						} else {
 							database.updateAnalysis(item, analysis, conn);
 						}
-						if(item.getFileCreationTime() <= 0) {
-							database.updateCreateTime(item, gpxFile.metadata.time);
+						if (item.getFileCreationTime() <= 0) {
+							database.updateCreateTime(item, GPXUtilities.getCreationTime(gpxFile));
```

Comment:
It will be better to save properly time in metadata, because now we have some places which use time from it, others use this new method